app/client/keybase/keyvault.go

```diff
@@ -0,0 +1,296 @@
+package keybase
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/pokt-network/pocket/shared/crypto"
+)
+
+// VaultKeybase implements the Keybase interface using HashiCorp Vault
+type VaultKeybase struct {
+	client *api.Client
+	mount  string
+}
+
+// NewVaultKeybase returns a new instance of VaultKeybase
+func NewVaultKeybase(client *api.Client, mount string) *VaultKeybase {
+	return &VaultKeybase{
+		client: client,
+		mount:  mount,
+	}
+}
+
+// Create new keypair entry in Vault
+func (vk *VaultKeybase) Create(passphrase, hint string) error {
+	keyPair, err := crypto.CreateNewKey(passphrase, hint)
+	if err != nil {
+		return err
+	}
+
+	keyPairBytes, err := json.Marshal(keyPair)
```

Comment:
@0xBigBoss Here the `KeyPair` struct actually has its own marshalling functions so you could do `keyPairBz, err := keyPair.Marshal()`

Reply:
@h5law originally i implemented it with this function, but I opted for JSON since it may need to be consumed by other SDK clients. I'm not an expert in go serialization and how portable the format is. Compare the two results. wdyt?

```
8��
EncKeyPair��	PublicKey
PrivKeyArmour����=github.com/pokt-network/pocket/shared/crypto.Ed25519PublicKey
" �~^9�2��|M��N�	j܁o�k�TN��V��({"kdf":"scrypt","salt":"9d812b1fcc5709c60b35dd132190cd95","secparam":"12","hint":"hint","ciphertext":"YH5R2dN1deUNO6NBAO0hXC5y3L8WuEgX3ur6FF5mNRdmwtvdxlkJxk2OeIZ5+DRHPcE6G6Lr0VlbbIC57PfnPB59UGX8Y9NWCefGutSe7gsdvQJxSA09R/qYs/XoxHAi896hacP6EVLtE7rfez7ZXAohvB+Ps1AeFAf8yCVtUHzqhMUJEPEQLt48LVHAj447"}
```

vs with json format

```json
{"PublicKey":"122004d236918c9a92bd5f21e02c33b4e6451b6d078796e6a0d7ea61d0867d0f","PrivKeyArmour":"{\"kdf\":\"scrypt\",\"salt\":\"bcfdd0c8eb441d33c27ce26cbbae3431\",\"secparam\":\"12\",\"hint\":\"hint\",\"ciphertext\":\"CPcnOHljTqgD9M6KBENZnL0uTLL8Vl0YAPREjh33KN6w+keGD/P3yknSTHyyyqoecgpZTx8KoAkGY44i46bggPhRoTbGLeb4baa5iMEbv3libXYxyeTfQiWVzLLN1hW6+F752nVh1R5o7t4UeiKciZmoT2XbPlZ4FGV50xQUKvts+tyrqjQCdkkZa0gkfzxY\"}"}
```

Reply:
@0xBigBoss `encoding/gob` is very specifc to golang - maybe for cross-language/SDK support it might be better to define a `keypair.proto` file? This would make the transmission a lot easier as the serialisation is universal across all languages. WDYT?

Reply:
1. Let's avoid exposing `gob` encoded bytes unless it lives in the scope of the project (i.e. looking at external services).

2. @h5law We can add a TODO in the keypair source code and implement it in the future. Definitely better for cross-compatibility.

3. The approach I'm thinking about in this use case is adding json tags to the keypair struct and adding a `MarshalJSON` function to the interface.

```go
type EncKeyPair struct {
	PublicKey     PublicKey  `json:"public_key"`
	PrivKeyArmour string    `json:"priv_key_armour"`
}
```

Usage would be:  `kpJson :=  kp.MarshalJSON()`